src/main/java/parser/FileParser.java

```diff
@@ -0,0 +1,59 @@
+package parser;
+
+import task.Task;
+import task.Todo;
+import task.Event;
+import task.Deadline;
+
+/**
+ * Represents a file parser that converts between strings and command objects.
+ */
+public class FileParser {
+
+    private String thisLine;
+
+    /**
+     * Constructor of the file parser.
+     *
+     * @param str the string stored in the file
+     */
+    public FileParser(String str) {
+        this.thisLine = str.trim();
+    }
+
+    /**
+     * Converts a string in the file to command objects.
+     *
+     * @return a task object depending on the content of the string
+     */
+    public Task lineToTask() {
+
+        String[] words = this.thisLine.split(" \\| ");
+
+        boolean isDone = false;
+        if (words[1].trim().equals("1")) {
+            isDone = true;
+        } else {
+            isDone = false;
+        }
+
+        String name = words[2].trim();
+        String duration = "";
+
+        String category = words[0].trim();
+        switch (category) {
```

Comment:
Good use of switch statement, allowing for more readable code. Perhaps you can consider enums too?